### PR TITLE
Avoid some unintended undefined behavior in devices/cpu/i386. Spotted by cppcheck

### PR DESCRIPTION
**Pull request by MoochMcGee**

### CODE REVIEW

File: src/devices/cpu/i386/i486ops.hxx [hunk 1]
```diff
@@ -284,7 +284,7 @@ void i386_device::i486_group0F01_16()      // Opcode 0x0f 01
 			}
 		case 6:         /* LMSW */
 			{
-				uint16_t b;
+				uint16_t b = 0;
```
Comment (cracyc) [inline src/devices/cpu/i386/i486ops.hxx]: Why would this be undefined?  b is set on both sides of the if statement.

**Comment by happppp** [comment]
Move "uint16_t b;" 2 lines further like it's done on line 407.

File: src/devices/cpu/i386/i386op32.hxx [hunk 2]
```diff
@@ -318,7 +318,7 @@ void i386_device::i386_bsr_r32_rm32()      // Opcode 0x0f bd
 	} else {
 		m_ZF = 0;
 		dst = temp = 31;
-		while( (src & (1 << temp)) == 0 ) {
+		while( (src & (1u << temp)) == 0 ) {
```
Comment (cuavas) [inline src/devices/cpu/i386/i386op32.hxx]: Please use uppercase `U` and `L` suffixes for literals - it's easy to visually confuse lowercase `l` with `1`, so we encourage uppercase for both.

File: src/devices/cpu/i386/i486ops.hxx (hunk 1 above)
Comment (cuavas) [reply src/devices/cpu/i386/i486ops.hxx]: As far as I can tell, the definition of `FAULT` doesn't use `b`.  Does it shut up the warning if you move the declaration of `b` after the GPF check?

**Comment by MoochMcGee** [comment]
Comments from both @cuavas and @cracyc addressed.

**Pull request merged**